registrations/permissions.py

```diff
@@ -52,3 +52,16 @@ def has_object_permission(self, request: Request, view, obj):
         if code != str(obj.cancellation_code):
             raise PermissionDenied(_("Cancellation code did not match"))
         return True
+
+
+class RegistrationUserRetrievePermission(permissions.BasePermission):
+    def has_permission(self, request, view):
+        # Only authenticated users can get object
+        return view.action == "retrieve" and request.user.is_authenticated
+
+    def has_object_permission(self, request: Request, view, obj):
+        user = request.user
+        if view.action != "retrieve":
+            return False
+
+        return len(obj.registration.registration_users.filter(email=user.email))
```

Comment:
```suggestion
        return obj.registration.registration_users.filter(email=user.email).exists()
```

Reply:
Fixed in https://github.com/City-of-Helsinki/linkedevents/pull/631/commits/11834ffbec8e2c83886750f3d7760a08351729cf